api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -118,6 +138,237 @@ type KubeletConfig struct {
+	// Maps to fs.inotify.max_user_watches.
+	// +kubebuilder:validation:Minimum=781250
+	// +kubebuilder:validation:Maximum=2097152
+	// +optional
+	FsInotifyMaxUserWatches *int32 `json:"fsInotifyMaxUserWatches,omitempty"`
+
+	// FsNrOpen specifies the maximum number of file-handles a process can allocate.
+	// Maps to fs.nr_open.
+	// +kubebuilder:validation:Minimum=8192
+	// +kubebuilder:validation:Maximum=20000500
+	// +optional
+	FsNrOpen *int32 `json:"fsNrOpen,omitempty"`
+
+	// KernelThreadsMax specifies the maximum number of all threads that can be created.
+	// Maps to kernel.threads-max.
+	// +kubebuilder:validation:Minimum=20
+	// +kubebuilder:validation:Maximum=513785
+	// +optional
+	KernelThreadsMax *int32 `json:"kernelThreadsMax,omitempty"`
+
+	// NetCoreNetdevMaxBacklog specifies maximum number of packets, queued on the INPUT side, when the interface receives packets faster than kernel can process them.
+	// Maps to net.core.netdev_max_backlog.
+	// +kubebuilder:validation:Minimum=1000
+	// +kubebuilder:validation:Maximum=3240000
+	// +optional
+	NetCoreNetdevMaxBacklog *int32 `json:"netCoreNetdevMaxBacklog,omitempty"`
+
+	// NetCoreOptmemMax specifies the maximum ancillary buffer size (option memory buffer) allowed per socket.
+	// Socket option memory is used in a few cases to store extra structures relating to usage of the socket.
+	// Maps to net.core.optmem_max.
+	// +kubebuilder:validation:Minimum=20480
+	// +kubebuilder:validation:Maximum=4194304
+	// +optional
+	NetCoreOptmemMax *int32 `json:"netCoreOptmemMax,omitempty"`
+
+	// NetCoreRmemDefault specifies the default receive socket buffer size in bytes.
+	// Maps to net.core.rmem_default.
+	// +kubebuilder:validation:Minimum=212992
+	// +kubebuilder:validation:Maximum=134217728
+	// +optional
+	NetCoreRmemDefault *int32 `json:"netCoreRmemDefault,omitempty"`
+
+	// NetCoreRmemMax specifies the maximum receive socket buffer size in bytes.
+	// Maps to net.core.rmem_max.
+	// +kubebuilder:validation:Minimum=212992
+	// +kubebuilder:validation:Maximum=134217728
+	// +optional
+	NetCoreRmemMax *int32 `json:"netCoreRmemMax,omitempty"`
+
+	// NetCoreSomaxconn specifies maximum number of connection requests that can be queued for any given listening socket.
+	// An upper limit for the value of the backlog parameter passed to the listen(2)(https://man7.org/linux/man-pages/man2/listen.2.html) function.
+	// If the backlog argument is greater than the somaxconn, then it's silently truncated to this limit.
+	// Maps to net.core.somaxconn.
+	// +kubebuilder:validation:Minimum=4096
+	// +kubebuilder:validation:Maximum=3240000
+	// +optional
+	NetCoreSomaxconn *int32 `json:"netCoreSomaxconn,omitempty"`
+
+	// NetCoreWmemDefault specifies the default send socket buffer size in bytes.
+	// Maps to net.core.wmem_default.
+	// +kubebuilder:validation:Minimum=212992
```

Comment:
Where did you find these defaults? Just curious.

Reply:
Defaults are mentioned in the [How-to guides of Linux OS Custom Config](https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration#linux-os-custom-configuration) by the field names.